### PR TITLE
Change to get latest block information in JSON format instead of multiple blocks

### PR DESCRIPTION
**Pull request by takafumi-hino**
Change to get latest block information in JSON format instead of multiple blocks.
Please review

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const exec     =  require('child_process').exec;
+
+const client = jayson.client.tcp({
+  port : 60401
+});
+
+
+function getBlockHash(height,callback){
+  let COMMAND = 'bitcoin-cli getblockhash ' + height;
+  exec(COMMAND, function(error, result){
```
Comment (rantan) [inline server.js]: You should use JSON-RPC via HTTP for calling Core's API.

Comment (rantan) [reply server.js]: Then, use `async/await` syntax.

**Comment by takafumi-hino** [comment]
Surpports URL input.

File: server.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  username : 'user',
+  password : 'password',
+  port     : 18443
+});
+
+function getBlock(blockHash, callback){
+  cl.getBlock(blockHash).then((result) => callback(result));
+};
+
+app.get('/blocks/:blockHash', (req, res) => {
```
Comment (takafumi-hino) [inline server.js]: Supports URL input.

File: package.json [hunk 3]
```diff
@@ -11,6 +11,8 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "basic-auth": "^2.0.1",
```
Comment (rantan) [inline package.json]: Is this package using in somewhere? If not, it should be removed.

File: server.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    let output = new Object();
+
+    let regex = new RegExp(/^[0-9a-fA-F]{64}$/);
+    let urlBlockHash = req.params.blockHash;
+
+    if(regex.test(urlBlockHash)){
```
Comment (rantan) [inline server.js]: I think returning quickly is better if it is error, like this

```javascript
if(!regex.test(urlBlockHash)){
  res.status(400).send('Bad request');
  return;
}

getBlock(...
```

File: server.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      getBlock(urlBlockHash, (blockInfo) => {
+        output.BlockHash             =  blockInfo.hash;
```
Comment (rantan) [inline server.js]: You can write object as literal like following code.

```javascript
const output = {
  blockHash: blockInfo.hash,
  noOfTransactions: blockInfo.nTx,
   ...
};
```

File: server.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+});
+
+function getBlock(blockHash, callback){
+  cl.getBlock(blockHash).then((result) => callback(result));
+};
+
+app.get('/blocks/:blockHash', (req, res) => {
+    let output = new Object();
+
+    let regex = new RegExp(/^[0-9a-fA-F]{64}$/);
```
Comment (rantan) [inline server.js]: It is better that using `const` when the variable will be never changed.

File: server.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    let urlBlockHash = req.params.blockHash;
+
+    if(regex.test(urlBlockHash)){
+      getBlock(urlBlockHash, (blockInfo) => {
+        output.BlockHash             =  blockInfo.hash;
+        output.No_of_Transactions    =  blockInfo.nTx;
```
Comment (rantan) [inline server.js]: `transaction count` or `ntx` is better.

File: server.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        output.Height                =  blockInfo.height;
+        output.Timestamp             =  Date(blockInfo.time*1000);
```
Comment (rantan) [inline server.js]: `blockInfo.time` is unix time, right?
It is better passing unix time data and parse it in client.

File: server.js (hunk 5 above)
Comment (rantan) [inline server.js]: Can you unify key names format as camel case or snake case?

File: package.json (hunk 3 above)
Comment (takafumi-hino) [reply package.json]: Did not use this package.
It has been deleted.

File: server.js (hunk 5 above)
Comment (takafumi-hino) [reply server.js]: decidedToBeBuiltWithALowerCamelCase.

File: server.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+function getBlock(blockHash, callback){
+  cl.getBlock(blockHash).then((result) => callback(result));
+};
+
+app.get('/blocks/:blockHash', (req, res) => {
+    const regex = new RegExp(/^[0-9a-fA-F]{64}$/);
+    let urlBlockHash = req.params.blockHash;
+
+    let output = new Object();
```
Comment (rantan) [inline server.js]: This is no longer needed now.

File: server.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+    if(!regex.test(urlBlockHash)){
+      res.status(400).send('Bad request');
+      return;
+    }
+    getBlock(urlBlockHash, (blockInfo) => {
+      const output ={
```
Comment (rantan) [inline server.js]: Can you add a space after `=`.

**Comment by takafumi-hino** [comment]
I fixed two point.
Confirm careful from next time.

**Pull request merged**